### PR TITLE
Add user timezone support for date filtering and aggregations

### PR DESCRIPTION
**Pull request by eswan18**
this required a *ton* of coaching and redirecting Claude, but ultimately this was a pretty tough feature that now appears to work perfectly...

## Summary

**Fixed incorrect date filtering and aggregations in the fitness dashboard.** Previously, runs were being filtered and aggregated based on UTC dates instead of the user's local timezone, causing wrong daily stats, training load calculations, and date range filters for users not in UTC.

For example, a user in Chicago selecting "July 15th" would incorrectly see runs that actually occurred on July 14th local time (but July 15th UTC), and miss runs from the evening of July 15th local time.

### The Core Problem
- **Date filtering**: Selecting date ranges used UTC dates instead of user's local timezone
- **Daily aggregations**: Mileage, TRIMP, and training load were bucketed by UTC date
- **Training metrics**: Training stress balance and burden calculations grouped runs incorrectly

### The Solution  
- **Automatic timezone detection**: Frontend detects user's timezone via browser API
- **Timezone-aware API calls**: All endpoints now accept optional `userTimezone` parameter
- **Precise datetime conversions**: Backend converts UTC run times to user's local timezone before filtering/aggregating
- **Correct date boundaries**: A 1 AM UTC run on July 15th now correctly appears on July 14th for Chicago users

### Frontend Changes
- Added `getUserTimezone()` utility for automatic timezone detection
- Updated all API fetch functions to pass user timezone parameter
- Modified React Query cache keys to include timezone for proper invalidation

### Backend Architecture Transformation
- **Eliminated confusion**: Removed `date` field from Run model - now only stores `datetime_utc`
- **Single source of truth**: All timezone conversion uses precise UTC datetime with explicit conversions
- **LocalizedRun model**: Created Run subclass with `localized_datetime` for clean timezone handling
- **Precise aggregations**: All daily stats now bucket runs by local date after timezone conversion

## Key Benefits

✅ **Accurate daily stats**: Users see correct mileage/TRIMP for each local day  
✅ **Proper date filtering**: Date range selections work as expected in user's timezone  
✅ **Correct training load**: ATL/CTL/TSB calculations use proper local date groupings  
✅ **Edge case handling**: Runs near midnight are correctly assigned to local dates  
✅ **No breaking changes**: API maintains backward compatibility via JSON serialization  

## Test Coverage

- **56 tests passing** with realistic datetime scenarios instead of artificial midnight UTC
- **Timezone edge cases**: Year boundaries, cross-timezone date assignment, late evening runs
- **Production usage**: Tests use same `LocalizedRun` conversion logic as aggregation functions

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: api/fitness/models/run.py [hunk 1]
```diff
@@ -43,8 +51,10 @@ def from_mmf(cls, mmf_run: MmfActivity) -> Self:
             if mmf_run.workout_date_utc is not None
             else mmf_run.workout_date
         )
+        # Create UTC datetime from the date (assuming start of day UTC)
+        workout_datetime_utc = datetime.combine(workout_date, datetime.min.time()).replace(tzinfo=timezone.utc).replace(tzinfo=None)
```
[the remark this answers is not in the document]
Comment (eswan18) [reply api/fitness/models/run.py:55]: sorta, but in the larger context of how we're passing this stuff around, I actually think naive datetime will be easiest
